subprojects/launcher/src/integTest/groovy/org/gradle/launcher/SystemClassLoaderTest.groovy

```diff
@@ -82,7 +84,19 @@ class SystemClassLoaderTest extends AbstractIntegrationSpec {
 
         lines.find { it == heading } // here for nicer output if the output isn't what we expect
         def headingIndex = lines.indexOf(heading)
-        lines[headingIndex + 1] == "1"
-        lines[headingIndex + 2].contains("gradle-launcher")
+        def classpathSize = Integer.parseInt(lines[headingIndex + 1])
+        // The gradle-launcher.jar is mandatory.
+        // The gradle-instrumentation-agent.jar may not be available if the build runs in no-daemon mode.
+        // There should not be more jars on the system classpath.
+        classpathSize == 1 || classpathSize == 2
+        def maybeHasAgent = classpathSize > 1
+
+        def libraries = lines[headingIndex + 2..<headingIndex + 2 + classpathSize]
+        libraries.any {
+            it.contains("gradle-launcher")
+        }
+        !maybeHasAgent || libraries.any {
```

Comment:
So it's not really a `maybe`, is it?

Reply:
`maybeHasAgent == true` means that we have a second jar on the classpath, which may be the agent (hence the prefix) or something else. So if it is present, we have to check.